tools/query_verification/src/main/java/com/google/bigquery/BigQueryManager.java

```diff
@@ -279,8 +281,55 @@ public String getName() {
             StandardSQLTypeName type = fields.get(i).getType().getStandardType();
 
             Object result;
-            // TODO Convert value to appropriate Java type
-            result = value.getStringValue();
+            try {
+                switch (type) {
+                    case BOOL:
+                        // returns boolean
+                        result = value.getBooleanValue();
+                        break;
+                    case FLOAT64:
+                        // returns double
+                        result = value.getDoubleValue();
+                        break;
+                    case INT64:
+                        // returns long
+                        result = value.getLongValue();
+                        break;
+                    case NUMERIC:
+                        // returns BigDecimal
+                        result = value.getNumericValue();
+                        break;
+                    case STRUCT:
+                        FieldList subFields = fields.get(i).getSubFields();
+                        FieldValueList subValues = value.getRecordValue();
+                        List<Object> subResults = parseResults(subValues, subFields);
+
+                        // returns List<Object>
+                        result = subResults;
+                        break;
+                    case DATE:
+                        // returns Date
+                        result = new SimpleDateFormat("yyyy-MM-dd").parse(value.getStringValue());
+                        break;
+                    case DATETIME:
+                        // returns Date
+                        result = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss.SSSSSS").parse(value.getStringValue());
+                        break;
+                    case TIME:
+                        // returns Date
+                        result = new SimpleDateFormat("hh:mm:ss.SSSSSS").parse(value.getStringValue());
+                        break;
+                    case TIMESTAMP:
+                        // returns Date
+                        result = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSSSSS zzz").parse(value.getStringValue());
+                        break;
+                    case STRING: default:
```

Comment:
I would recommend making these separate cases and leaving a comment that we choose to use the String representation if the type is not known otherwise.